script.js

```diff
@@ -0,0 +1,107 @@
+class BookShop {
+  discountRateByBookNumber ;
+  topRatedBooks;
+  setDiscountRateByBookNumber (promotions) {
+    this.discountRateByBookNumber = {...promotions}
+  }
+
+  setTopRatedBooks (books){
+    this.topRatedBooks = books
+  }
+
+  priceSummation (data) {
+    let result = []
+    let transaction = {...data}
+
+    while (Object.keys(transaction).length !== 0 ) {
+      let sum = []
+      for (let key in transaction){
+        if (transaction[key] == 0){
+          delete transaction[key]
+        } else {
+          sum.push(transaction[key])
+          transaction[key] -= 1
+        }
+      }
+      sum.length > 1 ? result.push(sum.length*80*this.discountRateByBookNumber[sum.length]) : result.push(sum.length*80)
```

Comment:
Alternatively, we could do like this.

```suggestion
     const basePrice = sum.length*80
     const newResult = sum.length > 1 
         ? basePrice*this.discountRateByBookNumber[sum.length] 
         : basePrice
     result.push(newResult)
```

Reply:
thats really cleaner krub p'Top